packages/flame/test/components/fps_component_test.dart

```diff
@@ -0,0 +1,50 @@
+import 'package:flame/components.dart';
+import 'package:flame_test/flame_test.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('FPSComponent', () {
+    testWithFlameGame('reports correct FPS for 1 frames', (game) async {
+      final fpsComponent = FPSComponent();
+      await game.ensureAdd(fpsComponent);
+      game.update(1 / 60);
+
+      expect(fpsComponent.fps, 60.0);
+    });
+
+    testWithFlameGame('reports correct FPS with full window', (game) async {
+      const windowSize = 30;
+      final fpsComponent = FPSComponent(windowSize: windowSize);
+      await game.ensureAdd(fpsComponent);
+      for (var i = 0; i < windowSize; i++) {
+        game.update(1 / 60);
+      }
+
+      expect(fpsComponent.fps, 60.0);
+    });
+
+    testWithFlameGame('reports correct FPS with slided window', (game) async {
+      const windowSize = 30;
+      final fpsComponent = FPSComponent(windowSize: windowSize);
+      await game.ensureAdd(fpsComponent);
+      for (var i = 0; i < 2 * windowSize; i++) {
+        game.update(1 / 60);
+      }
+
+      expect(fpsComponent.fps, 60.0);
+    });
+
+    testWithFlameGame('reports correct FPS with varying dt', (game) async {
+      const windowSize = 40;
+      final fpsComponent = FPSComponent(windowSize: windowSize);
+      await game.ensureAdd(fpsComponent);
+      for (var i = 0; i < 1.5 * windowSize; i++) {
+        // Alternating between 50 and 100 FPS
+        final dt = 1 / (100 / (1 + i % 2));
+        game.update(dt);
+      }
+
+      expect(fpsComponent.fps, 75.0);
```

Comment:
This should be around 67.
If one frame takes 0.01s, and next 0.02s, then 2 frames take 0.03s, and 66 frames take 0.99s. Add one more 0.01s frame -- and there were 67 frames in total within the last 1.00 second.